src/fields/addCounter.jsx

```diff
@@ -0,0 +1,82 @@
+import { Component } from 'react';
+
+export default function addCounter(TextInput) {
+
+  return class Wrapper extends Component {
+    constructor(props) {
+      super(props)
+
+      this.state = {
+        charCount: 0,
+        wordCount: 1,
+        value: ''
+      }
+
+      this.textInputProps = Object.assign({}, this.props);
+      ['charLimit', 'wordLimit', 'cutAtLimit', 'onChange'].forEach((item)=> delete this.textInputProps[item])
+    }
+
+    onChange(e) {
+      let value = e.target.value
+      let charLimit = this.props.charLimit
+      let wordLimit = this.props.wordLimit
+
+      if(charLimit) {
+
+        if(this.props.cutAtLimit && value.length > charLimit) {
+          this.setState({
+            charCount: charLimit,
+            value: value.substring(0, charLimit)
+          })
+        }
+        else {
+          this.setState({
+            charCount: value.length,
+            value: value
+          })
+        }
+
+      } else if(wordLimit) {
+
+        let words = value.split(' ')
+        if(this.props.cutAtLimit && words.length > wordLimit) {
+          this.setState({
+            wordCount: wordLimit,
+            value: words.slice(0, wordLimit).join(' ')
```

Comment:
if i use words for word count, the join will disrupt the formatting (two spaces will convert into one etc.) So i am taking no of spaces as word count